packages/datetime/src/datePickerCaption.tsx

```diff
@@ -123,25 +105,25 @@ export class DatePickerCaption extends React.PureComponent<IDatePickerCaptionPro
         this.positionArrows();
     }
 
-    private containerRefHandler = (r: HTMLElement) => (this.containerElement = r);
-
     private positionArrows() {
         // measure width of text as rendered inside our container element.
-        const monthWidth = Utils.measureTextWidth(
+        const monthTextWidth = measureTextWidth(
             this.displayedMonthText,
             Classes.DATEPICKER_CAPTION_MEASURE,
             this.containerElement,
         );
-        this.setState({ monthWidth });
+        const monthSelectWidth = this.containerElement.firstElementChild.clientWidth;
```

Comment:
reaching into first child element for client width? makes me nervous that we're not doing more null checks

Reply:
also this will trigger a layout